csrc/scheduler/utils.cpp

```diff
@@ -2234,17 +2234,37 @@ void propagateReshapeTransforms(Fusion* fusion, const ComputeAtMap& ca_map) {
     for (auto logical_id : tv->getLogicalDomain()) {
       if (terminating_reshape_dims.find(logical_id) !=
           terminating_reshape_dims.end()) {
+        // Check if logical ID is directly in the loop domain
         auto find_it = std::find(
             tv->getLoopDomain().begin(), tv->getLoopDomain().end(), logical_id);
+
+        // If not found directly and there is a sharded loop ID,
+        // check if the logical ID is the same as the producer of the DID split.
+        if (find_it == tv->getLoopDomain().end()) {
+          int64_t sharded_axis = getShardedLoopAxis(tv, ParallelType::DIDx);
+          if (sharded_axis != -1) {
+            // Get the split operation that created the DIDx dimension
+            auto split = dynamic_cast<Split*>(
+                tv->getLoopDomain().at(sharded_axis)->definition());
+            if (split && split->in() == logical_id) {
+              find_it = std::find(
+                  tv->getLoopDomain().begin(),
+                  tv->getLoopDomain().end(),
+                  split->inner());
+            }
+          }
+        }
+
         NVF_ERROR(
             find_it != tv->getLoopDomain().end(),
             "Require ",
             logical_id,
             " is in the active domain of ",
             tv->toString(),
             " for view propagation.");
-        int64_t old_pos = std::distance(tv->getLoopDomain().begin(), find_it);
 
+        // Reorder the reshape dimensions to the front of the domain
```

Comment:
@naoyam I'm quite confused by this pre-existing logic before I can understand this PR. Why is it necessary to move reshape dimensions to the front of the loop domain? It can cause conflict with the pre-existing assumption that DIDs have to be the front as well.

Reply:
I think this is related to the propagation done at line 2279. IIRC, it propagates the outermost N dimensions, where N is `old2new.size()` in this case. Since here we just want to propagate the transformations related to the rfactor, this is how we limit the propagation.

We can probably just reorder `tv` back after line 2279.